draft-cms-masque-ip-proxy-reqs.md

```diff
@@ -133,30 +143,33 @@ packets, in their unmodified entirety. The protocol will support both IPv6
 
 ## Maximum Transmission Unit
 
-The protocol will allow endpoints to negotiate the Maximum Transmission Unit
-(MTU) in use over a given Data Transport. This will allow avoiding IP
-fragmentation, especially as IPv6 does not allow IP fragmentation by nodes
+The protocol will allow endpoints to inform each other of the Maximum
+Transmission Unit (MTU) they are willing to forward. This will allow avoiding
+IP fragmentation, especially as IPv6 does not allow IP fragmentation by nodes
 along the path.
 
 ## IP Assignment
 
-Both the client or server may request to be assigned an IP address range. In
-response to the request, the peer will respond with an IP address range of its
-choosing.
+The server will be able to assign an IP address of its choosing to the client.
+The client may suggest a preference of IP address which the server can choose
+to follow. Similarly, to support the network-to-network use case, the client
+will be able to assign an IP address to the server, and the server may suggest
+a preference.
 
 ## Route Negotiation
 
 At any point in an IP Session (not limited to its initial negotiation), the
-protocol will allow both client and server to request routes to specific IP
-address ranges. In response to this request, the peer will have the ability to
-respond with a subset of routes that it is willing to accept, or deny the
-request.
+protocol will allow both client and server to inform its peer that it can route
+a set of IP prefixes. Both endpoints can also request a route to a given prefix,
```

Comment:
Is there much precedent for the request-a-route approach? I'd sort of envisioned both sides telling the other "here's the full set of addresses I'm willing to let you route to". If other addresses ranges later become routable, then the endpoint could push out another announcement.

Reply:
There's precedent in as much as that's how IKEv2 traffic selectors work. That approach allows the client to say "I'd like a route to ::/0" or "I'd like to route to this intranet" and the server can use that in determining what it'll let the client use

Reply:
I see. I think that could be useful in the Consumer VPN scenario you describe above where the client is asking for routes to public addresses that it's already aware of.

I'm trying to think through how naturally this approach fits when it needs to request a route into a private network and it has no knowledge of the private network's address space. How does the client know what ranges to request a route to in that scenario?

Reply:
In that scenario, the client doesn't request routes, it instead relies on the server unilaterally announcing them.

Reply:
That's reasonable. In the interest of not nit'ing too hard this early on, approved.

Reply:
Thanks! And I fully agree that we'll iterate on this text many times before this gets published.